script.js

```diff
@@ -155,12 +154,19 @@ function filterQuestions() {
 								}</span>`;
 							})
 							.join(" ")}
+							<hr />
+							<p class="explanation">${explanation}</p>
     					<button class="reveal-btn">Reveal</button>
 					</li>`;
 		})
 		.join(" ");
 
-	questionsList.innerHTML = listItem;
+	const noResult = `<p>Can not find anything. Please, try something else</p>
+`;
+
+	filteredQuestions.length
+		? (questionsList.innerHTML = listItem)
+		: (questionsList.innerHTML = noResult);
```

Comment:
Very nice edge case handling! You show a sensible message to the user even in case you have no data to show.